pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -161,36 +153,76 @@ func newClaimInfoCache(stateDir, checkpointName string) (*claimInfoCache, error)
 			entry.ResourceHandles,
 		)
 		for pluginName, cdiDevices := range entry.CDIDevices {
-			err := info.addCDIDevices(pluginName, cdiDevices)
+			err := info.updateCDIDevices(pluginName, cdiDevices)
 			if err != nil {
 				return nil, fmt.Errorf("failed to add CDIDevices to claimInfo %+v: %+v", info, err)
 			}
 		}
-		cache.add(info)
+		if err = cache.add(&info); err != nil {
+			return nil, fmt.Errorf("failed to add claimInfo %+v to the cache: %w", info, err)
+		}
 	}
 
 	return cache, nil
 }
 
-func (cache *claimInfoCache) add(res *ClaimInfo) {
+func (cache *claimInfoCache) add(claimInfo *ClaimInfo) error {
 	cache.Lock()
 	defer cache.Unlock()
 
-	cache.claimInfo[res.ClaimName+res.Namespace] = res
+	if _, ok := cache.claimInfo[claimInfo.ClaimName+claimInfo.Namespace]; ok {
+		return fmt.Errorf("claim %s, namespace %s: claim info already in cache", claimInfo.ClaimName, claimInfo.Namespace)
```

Comment:
The caller should (and does) print the details of the info struct passed in:
```suggestion
		return fmt.Errorf("already present")
```